packages/react-native-codegen/src/parsers/flow/modules/index.js

```diff
@@ -450,25 +450,21 @@ function translateFunctionTypeAnnotation(
           ),
         );
 
-      if (paramTypeAnnotation.type === 'VoidTypeAnnotation') {
-        throw new UnsupportedFunctionParamTypeAnnotationParserError(
-          hasteModuleName,
-          flowParam.typeAnnotation,
-          paramName,
-          'void',
-          language,
-        );
-      }
+      throwIfUnsupportedFunctionParamTypeAnnotationParserError(
+        hasteModuleName,
+        flowParam.typeAnnotation,
+        paramTypeAnnotation.type,
+        paramName,
+        'VoidTypeAnnotation',
+      );
 
-      if (paramTypeAnnotation.type === 'PromiseTypeAnnotation') {
-        throw new UnsupportedFunctionParamTypeAnnotationParserError(
-          hasteModuleName,
-          flowParam.typeAnnotation,
-          paramName,
-          'Promise',
-          language,
-        );
-      }
+      throwIfUnsupportedFunctionParamTypeAnnotationParserError(
+        hasteModuleName,
+        flowParam.typeAnnotation,
+        paramTypeAnnotation.type,
+        paramName,
+        'PromiseTypeAnnotation',
+      );
```

Comment:
What about saving some more lines (deleting line from 453 to 459)with this:
```suggestion
      const forbiddenTypes = new Set(['VoidTypeAnnotation', 'PromiseTypeAnnotation']);
      const currentType = paramTypeAnnotation.type;
      if (forbiddenTypes.has(currentType)) {
        throwIfUnsupportedFunctionParamTypeAnnotationParserError(
          hasteModuleName,
          flowParam.typeAnnotation,
          paramTypeAnnotation.type,
          paramName,
          currentType,
        );
      }
```

This is better because, if in the future we want to forbid more type, we can just work on the Set instead of make one call for each. 
Perhaps, with this approach, you may also be able to solve the Flow problem.

What do you think?

Reply:
I tried to use a Set but Flow couldn't infer that `paramTypeAnnotation.type` could not equal `VoidTypeAnnotation` or `PromiseTypeAnnotation` so I had the same Flow error when calling `wrappNullable` as before (https://github.com/facebook/react-native/pull/35057#issuecomment-1288170962).

The only solution I found is to replace the Set with multiple OR conditions and to add a `return`.